test/test_pipeline/test_tabular_classification.py

```diff
@@ -186,3 +192,26 @@ def test_get_fit_requirements(self, fit_dictionary):
         assert isinstance(fit_requirements, list)
         for requirement in fit_requirements:
             assert isinstance(requirement, FitRequirement)
+
+    def test_apply_search_space_updates(self, fit_dictionary, search_space_updates):
```

Comment:
I think this test is great, do not change it, but also add 2 additional tests:

- Take the whole configuration space, and convert it to dict, store it. Use it to create a new configuration space update where we only leave one choice dynamically for choices, and for uniform/int hyperparameters, make the range 0 and 1. Then get the configuration space using get configuration space, convert it to dict and make sure that the ranges change accordingly.

2. Put a broken configuration update so that it raises an error

Reply:
could you explain the first point? I think the .__choice__ hyperparameter is chosen using include exclude. Is that what you want to test? Both include and search space update run together?

Reply:
Yeah, test them both together but also want to make sure that every thing can be updated. For instance, there was a typo with one config space... The only way to catch this is if we create a test that changes every element of the config space and see if it changed. Makes sense?

Reply:
yeah I have added this in the latest push